app/controllers/calagator/sources_controller.rb

```diff
@@ -7,7 +7,8 @@ def import
     @importer = Source::Importer.new(params.permit![:source])
     respond_to do |format|
       if @importer.import
-        format.html { redirect_to events_path, flash: { success: render_to_string(layout: false) } }
+        redirect_target = @importer.events.count == 1 ? @importer.events.first : events_path
```

Comment:
Neat Ruby trick: `Enumerable` has a `#one?` method, so you could shorten the check to to `@importer.events.one?`. Not a showstopper by any means, just neat.